js/ui/validationMessage.js

```diff
@@ -0,0 +1,142 @@
+import $ from '../core/renderer';
+import registerComponent from '../core/component_registrator';
+import Overlay from './overlay';
+import { extend } from '../core/utils/extend';
+import { encodeHtml } from '../core/utils/string';
+import { getDefaultAlignment } from '../core/utils/position';
+
+const INVALID_MESSAGE = 'dx-invalid-message';
+const INVALID_MESSAGE_AUTO = 'dx-invalid-message-auto';
+const INVALID_MESSAGE_ALWAYS = 'dx-invalid-message-always';
+const INVALID_MESSAGE_CONTENT = 'dx-invalid-message-content';
+const VALIDATION_MESSAGE_MIN_WIDTH = 100;
+
+const ValidationMessage = Overlay.inherit({
+    _getDefaultOptions() {
+        return extend(this.callBase(), {
+            integrationOptions: {},
+            templatesRenderAsynchronously: false,
+            shading: false,
+            width: 'auto',
+            height: 'auto',
+            closeOnOutsideClick: false,
+            closeOnTargetScroll: false,
+            animation: null,
+            visible: true,
+            propagateOutsideClick: true,
+            _checkParentVisibility: false,
+            rtlEnabled: false,
+            contentTemplate: this._renderInnerHtml,
+            maxWidth: '100%',
+
+            mode: 'auto',
+            validationErrors: undefined,
+            positionRequest: undefined,
+            boundary: undefined,
+            offset: { h: 0, v: 0 }
+        });
+    },
+
+    _init() {
+        this.callBase();
+        this.updateMaxWidth();
+        this._updatePosition();
+    },
+
+    _initMarkup() {
+        this.callBase();
+
+        this.$element().addClass(INVALID_MESSAGE);
+        this._wrapper().addClass(INVALID_MESSAGE);
+        this._toggleModeClass();
+        this._updateContentId();
+    },
+
+    _updateContentId() {
+        const contentId = $(this.option('container')).attr('aria-describedby');
+
+        this.$content()
+            .addClass(INVALID_MESSAGE_CONTENT)
+            .attr('id', contentId);
+    },
+
+    _renderInnerHtml(element) {
+        const $element = element && $(element);
+        const validationErrors = this.option('validationErrors') || [];
+        let validationErrorMessage = '';
+        validationErrors.forEach((err) => {
+            const separator = validationErrorMessage ? '<br />' : '';
+            validationErrorMessage += separator + encodeHtml(err?.message || '');
+        });
+
+        $element?.html(validationErrorMessage);
+    },
+
+    _toggleModeClass() {
+        const mode = this.option('mode');
+        this.$element()
+            .toggleClass(INVALID_MESSAGE_AUTO, mode === 'auto')
+            .toggleClass(INVALID_MESSAGE_ALWAYS, mode === 'always');
+    },
+
+    updateMaxWidth() {
+        const target = this.option('target');
+        const targetWidth = target?.outerWidth?.() || $(target)?.outerWidth?.();
```

Comment:
Is there a reason to check `$(target)?.`?
![image](https://user-images.githubusercontent.com/1554153/93492385-699fda00-f913-11ea-8db2-1afd7b9da894.png)

Reply:
fixed